src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/UI/VsShellUtilitiesHelper.cs

```diff
@@ -30,9 +30,9 @@ public VsShellUtilitiesHelper(IProjectThreadingService threadingService)
         {
             await _threadingService.SwitchToUIThread();
 
-            IVsAppId vsAppId = await vsAppIdService.GetValueAsync();
+            IVsAppId? vsAppId = await vsAppIdService.GetValueAsync();
 
-            if (ErrorHandler.Succeeded(vsAppId.GetProperty((int)VSAPropID.VSAPROPID_ProductSemanticVersion, out object oVersion)) &&
+            if (ErrorHandler.Succeeded(vsAppId!.GetProperty((int)VSAPropID.VSAPROPID_ProductSemanticVersion, out object oVersion)) &&
```

Comment:
Whats the logic of making GetValueAsync() return a nullable, and then immediately using the damnit operator?

Reply:
This PR makes all the `GetService` calls (and related APIs) nullable. However we use those services in a _lot_ of places, and I don't think we are realistically going to add null checks everywhere, especially for services so integral to VS that if they're absent we've got bigger problems. The `!` here reflects that assumption.

Reply:
I get that.. but wouldn't it make more sense to mark the API as not nullable? If we don't expect people to null check, then shouldn't the stated return type reflect that?

Reply:
I guess it's subjective. There are cases where we do check whether the returned service is null. In future the analyzer may change to suggest removing redundant null checks, which would then break working code. My preference (at this early stage of using NRT) is to try and reflect reality in the type system, like the rest of the type system.